### PR TITLE
306 improved result parsing

### PR DESCRIPTION
**Pull request by Paul-Ferrell**

### CODE REVIEW

**Comment by Paul-Ferrell** [comment]
Fixes #306

**Comment by Paul-Ferrell** [comment]
Closes #306

File: docs/tutorials/extracting_results.rst [hunk 1]
```diff
@@ -0,0 +1,510 @@
...
+                  "node3": {"flops": 39.49},
+                  "node4": {"flops": 139.72},
+                  "node5": {"flops": 31.67}},
+     "result": "PASS",
+     "return_value": 0,
+    }
+
+**TASK:** Add a parser for the 'accel' data in the per-node results.
+
+Wild Evaluations
```
Comment (nicholas-sly) [inline docs/tutorials/extracting_results.rst:398]: Is this supposed to be 'Wildcard Evaluations'?

File: docs/tutorials/extracting_results.rst [hunk 2]
```diff
@@ -0,0 +1,510 @@
...
+            # Extract just the list of outlier names.
+            outliers: 'keys(_outliers)'
+
+**TASK:** Add these evaluations to your config and test them.
+
+Constants
+---------
+
+Sometimes the value you need to add to the results is already in a Pavilion
+variable, or
```
Comment (nicholas-sly) [inline docs/tutorials/extracting_results.rst:456]: You wandered off again...

**Review by nicholas-sly — CHANGES_REQUESTED** [review]
Looked over the documentation section.  Let me know if you have any questions.

File: docs/results/basics.rst [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+               "min_cpus": "36",
+               "min_mem": "131933.144",
+               "total_mem": "125821"},
+     "started": "2020-07-06 15:28:02.398033",
+     "sys_name": "pav-test",
+     "per_file": {
+        "node01": {"raw_speed": 33},
+        "node03": {"raw_speed": 39},
+        "node05": {"raw_speed": 42}
+     }
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/results/basics.rst:81]: If this were python, yes. But it's JSON, and a trailing comma is a syntax error.

File: docs/results/evaluate.rst [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+    {
+        "name": "example_results",
+        "table":
+            "node1": {"speed": 30.0, "items": 5},
+            "node2": {"speed": 50.0, "items": 4},
+            "node3": {"speed": 70.0, "items": 3}
+        "sched": {
+            "test_nodes": 3,
+            "test_node_list": ["node1", "node2", "node3"],
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/results/evaluate.rst:42]: and other things.

File: docs/results/evaluate.rst [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            # 'num/2'
+            # This will then be evaluated, and the 'num' result value will be
+            # used (50).
+            # The result 'half_num' key will thus be 25. (ie: ``50/2`` )
+
+            # If we want to include a Pavilion variable as a string, it must
+            # be put in quotes.
+            msg_len: 'len("{{message}}")'
+            # This will become: 'len("Hiya")'
+            # Which will evaluate to ``4``.
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/results/evaluate.rst:97]: :+1:

File: docs/results/parse.rst [hunk 6]
```diff
@@ -0,0 +1,537 @@
...
+.. code-block:: yaml
+
+    result_parse:
+        regex:
+            # When you use multiple groupings in a regex, the
+            # matches are returned in a list.
+            "speed, runtime, points":
+                regex: 'results: ([0-9.]+) ([0-9.]+) (\d+)'
+
+            "
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/results/parse.rst:127]: For reminding me that I needed another example there, apparently.

File: docs/results/parse.rst [hunk 7]
```diff
@@ -0,0 +1,537 @@
...
+    {
+      "fn": {
+        "node1": {"huge_size": null},
+        "node2": {"huge_size": "2M"},
+        "node3": {"huge_size": "4K"},
+        "node4": {"huge_size": null}
+      }
+    }
+
+- When using the **fullname** *per\_file* setting, the key cannot be
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/results/parse.rst:423]: Yep, legacy naming. Fixed it.

File: docs/tutorials/extracting_results.rst [hunk 8]
```diff
@@ -0,0 +1,510 @@
...
+    Good night!
+
+You can view these with ``pav cat <test_id> build/node5.out``.
+
+**TASK:** Run the test and view the run log and one of the node output files.
+
+Basic Regex Parsing
+-------------------
+
+There's a lot to parse here.
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/tutorials/extracting_results.rst:104]: Worth it.

File: docs/tutorials/extracting_results.rst [hunk 9]
```diff
@@ -0,0 +1,510 @@
...
+                    regex: 'Ran for: (\d+\.\d+)s'
+                _gflops:
+                    regex: 'GFlops: (\d+)'
+
+        # The 'result_evaluate' section lets us perform math. The syntax is
+        # the same the inside of '{{  }}' sections in Pavilion strings,
+        # but the *variables* are result keys! So we can reference '_gflops'
+        # here.
+        result_evaluate:
+            pflops: 'gflops/1000'
```
[the remark this answers is not in the document]
Comment (Paul-Ferrell) [reply docs/tutorials/extracting_results.rst:182]: :facepalm:

**Review by francinelapid — CHANGES_REQUESTED** [review]

**Comment by francinelapid** [comment]
Forgot to add this to my review:
- looked through code 
- updated result parser tests look better

**Pull request merged**